.github/workflows/release.yml

```diff
@@ -0,0 +1,76 @@
+name: release
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+      - "VERSION"
+
+jobs:
+  build:
+    name: Build and publish new release
+    runs-on: "ubuntu-latest"
+    if: github.repository == 'drivendataorg/deon'
+
+    steps:
+
+    - uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r dev-requirements.txt
+
+    - name: Build package
+      id: build_package
+      run: |
+        cat VERSION
+        echo "::set-output name=version::$(cat VERSION)"
+        make package
+
+    - name: Publish to Test PyPI
+      uses: pypa/gh-action-pypi-publish@v1.3.0
+      with:
+        user: ${{ secrets.PYPI_TEST_USERNAME }}
+        password: ${{ secrets.PYPI_TEST_PASSWORD }}
+        repository_url: ${{ secrets.PYPI_TEST_REPOSITORY }}
```

Comment:
Loving this secrets injection! (Also, I guess repository and username don't need to be secrets).